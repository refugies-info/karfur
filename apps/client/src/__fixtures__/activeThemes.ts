import { Id } from "@refugies-info/api-types";

export const activeThemesMock = [
  {
    colors: {
      color100: "#3D2884",
      color80: "#705FA4",
      color60: "#A396C4",
      color40: "#D6CDE4",
      color30: "#EFE8F4"
    },
    icon: {
      secure_url: "/images/icon_elearning.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Français.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/francais.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/francais.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-francais.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7df5" as Id,
    name: {
      fr: "apprendre le français",
      en: "learn French",
      ar: "تعلّم الفرنسيّة",
      fa: "آموختن زبان فرانسوی",
      ps: "فرانسوي زده کړئ",
      ru: "учить французский",
      uk: "Курси французької мови",
      ti: "ትምህርቲ ቓንቓ ፈርንሳይ ምምሃር"
    },
    "short": {
      fr: "Français",
      en: "French",
      ar: "الفرنسيّة",
      fa: "فرانسوی",
      ps: "فرانسوی",
      ru: "Французский",
      uk: "Французька мова",
      ti: "ፈረንሳይ"
    },
    position: 2,
    notificationEmoji: "🇫🇷",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#443023",
      color80: "#715F56",
      color60: "#9D8F88",
      color40: "#CABEBB",
      color30: "#EEE7E5"
    },
    icon: {
      secure_url: "/images/icon_office.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Administratif.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/admin.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/admin.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-administratif.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7df6" as Id,
    name: {
      fr: "gérer mes papiers",
      en: "manage my paperwork",
      ar: "إدارة أوراقي",
      fa: "مدیریت پرونده هایم",
      ps: "خپلې اسناد تنظیم کړئ",
      ru: "разобраться с документами",
      uk: "мої документи",
      ti: "ባዕለይ የሳልጥ  ወረቃቕተይ"
    },
    "short": {
      fr: "Administratif",
      en: "Admin",
      ar: "المعاملات الاداريّة",
      fa: "اداری",
      ps: "اداري",
      ru: "Административное",
      uk: "Адміністративні питання",
      ti: "ቤት ጽሕፈታት"
    },
    position: 1,
    notificationEmoji: "📝",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#06508C",
      color80: "#1E9ED1",
      color60: "#88C6E3",
      color40: "#C1E3F2",
      color30: "#DDF2F9"
    },
    icon: {
      secure_url: "/images/icon_house.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Logement.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/logement.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/logement.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-logement.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7df7" as Id,
    name: {
      fr: "me loger",
      en: "find housing",
      ar: "إيجاد سكن",
      fa: "پیدا کردن جای سکونت",
      ps: "استوګنځای پیداکړئ",
      ru: "найти жилье",
      uk: "Житло",
      ti: "ክነብር"
    },
    "short": {
      fr: "Logement",
      en: "Housing",
      ar: "إسكان",
      fa: "مسکن",
      ps: "استوګنځای",
      ru: "Жилье",
      uk: "Житло",
      ti: "መንበሪ ገዛ"
    },
    position: 5,
    notificationEmoji: "🏡",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#CD5605",
      color80: "#F97319",
      color60: "#FB9B5A",
      color40: "#FDD7BD",
      color30: "#FEEBDE"
    },
    icon: {
      secure_url: "/images/icon_bus.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Mobilité.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/transport.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/transport.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-mobilite.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7df8" as Id,
    name: {
      fr: "me déplacer",
      en: "move about",
      ar: "التنقّل",
      fa: "سفر کردن",
      ps: "تقل او حرکت",
      ru: "перемещаться",
      uk: "Транспорт",
      ti: "ኣነ ክሳፈር \\ ይጉዓዝ"
    },
    "short": {
      fr: "Mobilité",
      en: "Mobility",
      ar: "التنقّل",
      fa: "تحرک",
      ps: "خوځښت",
      ru: "Мобильность",
      uk: "Транспорт",
      ti: "ብቅዓት = ክእለት"
    },
    position: 7,
    notificationEmoji: "🚗",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#055E5A",
      color80: "#27A8A5",
      color60: "#82F2DB",
      color40: "#D7FAF2",
      color30: "#EDFDF9"
    },
    icon: {
      secure_url: "/images/icon_briefcase.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Insertion-pro.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/travail.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/travail.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-insertion.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7df9" as Id,
    name: {
      fr: "trouver un travail",
      en: "find a job",
      ar: "ايجاد عمل",
      fa: "پیدا کردن شغل",
      ps: "دنده ومومئ",
      ru: "найти работу",
      uk: "Працевлаштування",
      ti: "ሓደ ስራሕ ምርካብ"
    },
    "short": {
      fr: "Insertion pro",
      en: "Job hunting",
      ar: "الاندماج المهني",
      fa: "درج حرفه ای",
      ps: "مسلکي مرسته",
      ru: "Работа",
      uk: "Працевлаштування",
      ti: "ምትእትታው ሞያ ብውሕልልነት"
    },
    position: 4,
    notificationEmoji: "💼",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#910940",
      color80: "#D64B71",
      color60: "#E78CA2",
      color40: "#F7CDD4",
      color30: "#FFEDED"
    },
    icon: {
      secure_url: "/images/icon_heartBeat.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Santé.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/sante.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/sante.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-sante.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dfa" as Id,
    name: {
      fr: "me soigner",
      en: "get medical care",
      ar: "العناية الصحّية",
      fa: "درمان",
      ps: "خپله درملنه وکړئ",
      ru: "получить медицинскую помощь",
      uk: "Медичні послуги",
      ti: "ኣነ ክሕከም"
    },
    "short": {
      fr: "Santé",
      en: "Health",
      ar: "الصحّة",
      fa: "صحت",
      ps: "روغتیا",
      ru: "Здоровье",
      uk: "Охорона здоров'я",
      ti: "ጥዕና"
    },
    position: 6,
    notificationEmoji: "👨\u200d⚕️",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#095411",
      color80: "#26B759",
      color60: "#82E59B",
      color40: "#D4F4D7",
      color30: "#E8F4E8"
    },
    icon: {
      secure_url: "/images/icon_measure.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Formation-pro.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/metier.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/metier.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-formation-pro.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dfb" as Id,
    name: {
      fr: "apprendre un métier",
      en: "learn a job",
      ar: "تعلم مهنة",
      fa: "آموختن یک کسب",
      ps: "یو کسب زده کړئ",
      ru: "получить профессию",
      uk: "Професійна підготовка",
      ti: "ሓደ ሞያ ምምሃር"
    },
    "short": {
      fr: "Formation pro",
      en: "Professional training",
      ar: "التدريب المهني",
      fa: "آموزش حرفه ای",
      ps: "مسلکي روزنه",
      ru: "Профессиональная подготовка",
      uk: "Професійна освіта",
      ti: "ምስልጣን ሞያ"
    },
    position: 3,
    notificationEmoji: "👩🏼\u200d🏫",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#204F01",
      color80: "#47AA0A",
      color60: "#70CE33",
      color40: "#DFFDAD",
      color30: "#F2F9E5"
    },
    icon: {
      secure_url: "/images/icon_glasses.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Études.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/etudes.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/etudes.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-etudes.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dfc" as Id,
    name: {
      fr: "faire des études",
      en: "study",
      ar: "الدّراسة",
      fa: "شروع کردن تحصیل",
      ps: "زده کړې ترسره کړئ",
      ru: "учиться",
      uk: "Навчання",
      ti: "ትምህርቲ ምግባር"
    },
    "short": {
      fr: "Études",
      en: "Studies",
      ar: "الدّراسة",
      fa: "تحصیلات",
      ps: "زده کړې",
      ru: "Учеба",
      uk: "Навчання",
      ti: "ምምሃር"
    },
    position: 9,
    notificationEmoji: "🎓",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#073099",
      color80: "#1C66DD",
      color60: "#5EA5EF",
      color40: "#C4E8F8",
      color30: "#EEF8FF"
    },
    icon: {
      secure_url: "/images/icon_flag.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Bénévolat.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/benevolat.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/benevolat.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-benevolat.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dfd" as Id,
    name: {
      fr: "aider une association",
      en: "help an association",
      ar: "مساعدة جمعية",
      fa: "کمک انجمن",
      ps: "یوې ټولنې سره مرسته وکړئ",
      ru: "стать волонтером",
      uk: "Допомога ассоціації",
      ti: "ሓገዝ ንሓንቲ ማሕበር"
    },
    "short": {
      fr: "Bénévolat",
      en: "Volunteering",
      ar: "التطوع",
      fa: "داوطلب",
      ps: "داوطلب",
      ru: "Волонтерство",
      uk: "Волонтерство",
      ti: "ወልንታዊ"
    },
    position: 10,
    notificationEmoji: "🤝",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#770649",
      color80: "#D01F67",
      color60: "#FC6EB5",
      color40: "#FFC2E5",
      color30: "#FFEAF4"
    },
    icon: {
      secure_url: "/images/icon_couple.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Rencontre.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/rencontres.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/rencontres.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-rencontre.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dfe" as Id,
    name: {
      fr: "rencontrer des gens",
      en: "meet people",
      ar: "مقابلة أشخاص",
      fa: "تماس شدن با مردم",
      ps: "خلکو سره ووینئ",
      ru: "познакомиться с людьми",
      uk: "Зустрічі",
      ti: "ምስ ደቂ ሰባት ምርኻብ"
    },
    "short": {
      fr: "Rencontre",
      en: "Socialising",
      ar: "لقاء الآخرين",
      fa: "ملاقات",
      ps: "لیدنه",
      ru: "Общение",
      uk: "Зустрічі",
      ti: "ምርኻብ ወይ ምዝታይ"
    },
    position: 8,
    notificationEmoji: "💬",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#600566",
      color80: "#9E2DAD",
      color60: "#D16DEA",
      color40: "#EDC4FC",
      color30: "#F9F1FD"
    },
    icon: {
      secure_url: "/images/icon_soccer.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Loisir.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/loisirs.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/loisirs.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-loisirs.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7dff" as Id,
    name: {
      fr: "occuper mon temps libre",
      en: "occupy my free time",
      ar: "إشغال وقت فراغي",
      fa: "مشغول شدن در اوقات بیکار",
      ps: "خپل اضافی او فارغ وخت ډک کړئ",
      ru: "занять свободное время",
      uk: "Вільний час",
      ti: "ኣብ ትርፊ ግዘይ መደብ ምሓዝ"
    },
    "short": {
      fr: "Loisir",
      en: "Leisure",
      ar: "التّسلية",
      fa: "سرگرمی ها",
      ps: "تفریح",
      ru: "Досуг",
      uk: "Відпочинок",
      ti: "ምዝንጋዕ"
    },
    position: 11,
    notificationEmoji: "⚽",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#9E180A",
      color80: "#E05A3E",
      color60: "#EF9887",
      color40: "#FFD5D0",
      color30: "#F9E8E8"
    },
    icon: {
      secure_url: "/images/icon_triumph.svg",
      public_id: "",
      imgId: ""
    },
    banner: {
      secure_url: "/images/illustration_Culture.svg",
      public_id: "",
      imgId: ""
    },
    appBanner: {
      secure_url: "/images/app-banner/culture.png",
      public_id: "",
      imgId: ""
    },
    appImage: {
      secure_url: "/images/app/culture.svg",
      public_id: "",
      imgId: ""
    },
    shareImage: {
      secure_url: "/images/share/ri-culture.png",
      public_id: "",
      imgId: ""
    },
    _id: "6319f6b363ab2bbb162d7e00" as Id,
    name: {
      fr: "découvrir la culture",
      en: "discover French culture",
      ar: "اكتشاف الثقافة",
      fa: "اشناشدن با فرهنگ",
      ps: "د کلتور په اړه پوهاوی ترلاسه کړئ",
      ru: "узнать о культуре",
      uk: "Знайомство з культурою",
      ti: "ሓድሽ ባህሊ ምልላይ"
    },
    "short": {
      fr: "Culture",
      en: "Culture",
      ar: "الثقافة",
      fa: "فرهنگ",
      ps: "کلتور",
      ru: "Культура",
      uk: "Культура",
      ti: "ባህሊ"
    },
    position: 12,
    notificationEmoji: "🏰",
    active: true,
    adminComments: ""
  },
  {
    colors: {
      color100: "#77057A",
      color80: "#9E2DAD",
      color60: "#D16DEA",
      color40: "#DB8EEF",
      color30: "#FDC4FC"
    },
    icon: {
      secure_url: "https://res.cloudinary.com/dlmqnnhp6/image/upload/v1663058487/pictures/o52hghafc04bexxedrik.svg",
      public_id: "pictures/o52hghafc04bexxedrik",
      imgId: "63204239dd70acd9199aed06"
    },
    banner: {
      secure_url: "https://res.cloudinary.com/dlmqnnhp6/image/upload/v1662990767/pictures/o1x9jwn7ncwx18mqcnx9.svg",
      public_id: "pictures/o1x9jwn7ncwx18mqcnx9",
      imgId: "631f39b068917b340a0411fe"
    },
    appBanner: {
      secure_url: "https://res.cloudinary.com/dlmqnnhp6/image/upload/v1662990785/pictures/ninp4ko4vbnrorsgyz7z.svg",
      public_id: "pictures/ninp4ko4vbnrorsgyz7z",
      imgId: "631f39c368917b340a041202"
    },
    appImage: {
      secure_url: "https://res.cloudinary.com/dlmqnnhp6/image/upload/v1662999880/pictures/ot6qwuszncts5ppxusds.svg",
      public_id: "pictures/ot6qwuszncts5ppxusds",
      imgId: "631f5d4ad449751db1492051"
    },
    shareImage: {
      secure_url: "https://res.cloudinary.com/dlmqnnhp6/image/upload/v1662995793/pictures/wndug5571lzvcs1sguc1.png",
      public_id: "pictures/wndug5571lzvcs1sguc1",
      imgId: "631f4d52d4047aca79ffd170"
    },
    _id: "631f3bfe68917b340a04120c" as Id,
    name: {
      fr: "famille",
      uk: "Сім‘я",
      ps: "کورنۍ",
      fa: "خانواده",
      en: "Family",
      ti: "ስድራ-ቤት",
      ru: "Семья",
      ar: "العائلة"
    },
    "short": {
      fr: "Famille",
      uk: "Сім‘я",
      ps: "کورنۍ",
      fa: "خانواده",
      en: "Family",
      ti: "ስድራ-ቤት",
      ru: "Семья",
      ar: "العائلة"
    },
    position: 14,
    notificationEmoji: "👨\u200d👩\u200d👦",
    created_at: "2022-09-12T14:02:38.447Z",
    updatedAt: "2022-09-13T08:41:38.952Z",
    __v: 0,
    active: true,
    adminComments: ""
  }
]
