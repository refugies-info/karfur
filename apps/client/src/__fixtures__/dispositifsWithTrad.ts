import {
  ContentType,
  GetDispositifsWithTranslationAvancementResponse,
  TraductionsStatus,
} from "@refugies-info/api-types";

const content1 = {
  avancementTrad: 700,
  avancementValidation: 0,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "Demander un logement social (HLM)",
  titreMarque: "",
  tradStatus: TraductionsStatus.TO_REVIEW,
  type: ContentType.DEMARCHE,
  _id: "id1",
};

const content2 = {
  avancementTrad: 812,
  avancementValidation: 406,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "T2",
  titreMarque: "TM2",
  tradStatus: TraductionsStatus.PENDING,
  type: ContentType.DISPOSITIF,
  _id: "id2",
};
const content3 = {
  avancementTrad: 0,
  avancementValidation: 0,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "T3",
  titreMarque: "TM3",
  tradStatus: TraductionsStatus.TO_TRANSLATE,
  type: ContentType.DISPOSITIF,
  _id: "id3",
};

const content4 = {
  avancementTrad: 812,
  avancementValidation: 0,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "T4",
  titreMarque: "TM4",
  tradStatus: TraductionsStatus.VALIDATED,
  type: ContentType.DEMARCHE,
  _id: "id4",
};

const content5 = {
  avancementTrad: 812,
  avancementValidation: 0,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "T5",
  titreMarque: "TM5",
  tradStatus: TraductionsStatus.TO_TRANSLATE,
  type: ContentType.DEMARCHE,
  _id: "id5",
};

const content6 = {
  avancementTrad: 700,
  avancementValidation: 0,
  created_at: new Date("2019-11-08T10:53:52.398Z"),
  lastTradUpdatedAt: 1612514769665,
  nbMots: 812,
  titreInformatif: "TI6",
  titreMarque: "TM6",
  tradStatus: TraductionsStatus.TO_REVIEW,
  type: ContentType.DISPOSITIF,
  _id: "id6",
};

export const dispositifsWithTranslations: GetDispositifsWithTranslationAvancementResponse[] = [
  content1,
  content2,
  content3,
  content4,
  content5,
  content6,
];
