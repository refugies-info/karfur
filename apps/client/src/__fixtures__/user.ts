import { GetUserInfoResponse, RoleName, UserStatus } from "@refugies-info/api-types";

export const testUser: GetUserInfoResponse = {
  username: "test",
  _id: "testObjectId",
  structures: [],
  contributions: [],
  phone: "",
  roles: [],
  email: "",
  selectedLanguages: [],
  status: UserStatus.ACTIVE,
  sso: false,
};

export const testUserWithRoles: GetUserInfoResponse = {
  ...testUser,
  structures: ["testObjectId"],
  roles: [
    { nom: RoleName.ADMIN, _id: "testObjectId", nomPublique: "Admin" },
    { nom: RoleName.TRAD, _id: "testObjectId", nomPublique: "Trad" },
    {
      nom: RoleName.EXPERT_TRAD,
      _id: "testObjectId",
      nomPublique: "ExpertTrad",
    },
    {
      nom: RoleName.CONTRIB,
      _id: "testObjectId",
      nomPublique: "Contrib",
    },
  ],
};
