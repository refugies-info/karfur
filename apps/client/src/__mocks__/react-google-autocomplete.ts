/*** Mock Google Maps JavaScript API ***/
export const setupGoogleMock = () => {
  const google = {
    maps: {
      places: {
        AutocompleteService: class {},
        PlacesService: class {},
        PlacesServiceStatus: {
          INVALID_REQUEST: "INVALID_REQUEST",
          NOT_FOUND: "NOT_FOUND",
          OK: "OK",
          OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
          REQUEST_DENIED: "REQUEST_DENIED",
          UNKNOWN_ERROR: "UNKNOWN_ERROR",
          ZERO_RESULTS: "ZERO_RESULTS",
        },
      },
      Geocoder: () => {},
      GeocoderStatus: {
        ERROR: "ERROR",
        INVALID_REQUEST: "INVALID_REQUEST",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },
  };
  global.window.google = google;
};
