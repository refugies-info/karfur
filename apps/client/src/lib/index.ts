export const jsUcfirst = (string: string | null | undefined) =>
  (string && string.length > 1 && string.charAt(0).toUpperCase() + string.slice(1, string.length)) || "";

export const jsLcfirst = (string: string) =>
  string && string.length > 1 && string.charAt(0).toLowerCase() + string.slice(1, string.length);

export const limitNbCaracters = (string: string, nbCaractersMax: number) =>
  string.substring(0, Math.min(string.length, nbCaractersMax)) + (string.length > nbCaractersMax ? "..." : "");

export const removeAccents = (str = "") => {
  var accents = "ÀÁÂÃÄÅàáâãäåÒÓÔÕÕÖØòóôõöøÈÉÊËèéêëðÇçÐÌÍÎÏìíîïÙÚÛÜùúûüÑñŠšŸÿýŽž";
  var accentsOut = "AAAAAAaaaaaaOOOOOOOooooooEEEEeeeeeCcDIIIIiiiiUUUUuuuuNnSsYyyZz";
  const chars = str.split("");
  let i, x;
  for (i = 0; i < chars.length; i++) {
    if ((x = accents.indexOf(chars[i])) !== -1) {
      chars[i] = accentsOut[x];
    }
  }
  return chars.join("");
};
