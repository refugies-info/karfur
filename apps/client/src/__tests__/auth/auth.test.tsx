import { initialMockStore } from "@/__fixtures__/reduxStore";
import "jest-styled-components";
import { wrapWithProvidersAndRenderForTesting } from "../../../jest/lib/wrapWithProvidersAndRender";
import { setupGoogleMock } from "../../__mocks__/react-google-autocomplete";
import auth from "../../pages/auth";

jest.mock("next/router", () => require("next-router-mock"));

describe("auth", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    setupGoogleMock();
  });

  it("renders auth", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: auth,
      reduxState: {
        ...initialMockStore,
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
