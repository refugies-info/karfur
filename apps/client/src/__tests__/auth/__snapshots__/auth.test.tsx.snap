// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth renders auth 1`] = `
<DocumentFragment>
  <div
    class="container half"
  >
    <button
      class="fr-btn fr-btn--tertiary fr-btn--sm fr-icon-arrow-left-line fr-btn--icon-left"
      id="fr-button-:r0:"
    >
      Retour
    </button>
    <div
      class="title"
    >
      <h1>
        Bienvenue !
      </h1>
      <p
        class="subtitle"
      >
        Votre compte Réfugiés.info vous permet d’avoir une expérience personnalisée.
      </p>
    </div>
    <form>
      <div
        class="fr-input-group"
      >
        <label
          class="fr-label"
          for="input-:r1:"
        >
          Adresse mail
        </label>
        <input
          aria-describedby="input-:r1:-desc-error"
          class="fr-input"
          id="input-:r1:"
          type="email"
          value=""
        />
      </div>
      <button
        class="fr-btn fr-icon-arrow-right-line fr-btn--icon-right button"
        id="fr-button-:r2:"
        type="submit"
      >
        Continuer
      </button>
    </form>
    <div
      class="separator mx"
    >
      <span>
        ou connectez-vous avec
      </span>
    </div>
    <button
      class="fr-btn fr-btn--tertiary button mb-4"
      id="fr-button-:r3:"
    >
      <img
        alt=""
        class="me-2"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
      Google
    </button>
    <button
      class="fr-btn fr-btn--tertiary button mb-4"
      id="fr-button-:r4:"
    >
      <img
        alt=""
        class="me-2"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
      <img
        alt=""
        class="me-2"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
      Microsoft ou Outlook
    </button>
  </div>
</DocumentFragment>
`;
